jax/_src/pjit.py

```diff
@@ -1378,7 +1378,8 @@ def _pjit_lower(
   da = _fast_path_get_device_assignment(it.chain(in_shardings, out_shardings))
   in_shardings = SameDeviceAssignmentTuple(in_shardings, da)
   out_shardings = SameDeviceAssignmentTuple(out_shardings, da)
-  return _pjit_lower_cached(jaxpr, in_shardings, out_shardings, *args, **kwargs)
+  return _pjit_lower_cached(jaxpr, in_shardings, out_shardings, *args, **kwargs,
+                            lowering_platform_override=mlir.thread_local_state.lowering_platform_override)
```

Comment:
I don't think we want the compilation cache to depend on global variables like this.

In general, this is basically leaking abstraction across the pjit/jit and jax2tf boundary. pjit already has ways to tell to recompile if the platform changes via multiple routes.

Maybe we can talk about this?

Reply:
The lowering cache (not the compilation cache) must depend on the platform for which we are lowering.

Yes, we should talk.